ui-manchette-with-spacetimechart/src/helpers.ts

```diff
@@ -0,0 +1,166 @@
+import type {
+  OperationalPointType,
+  ProjectPathTrainResult,
+  StyledOperationalPointType,
+} from '@osrd-project/ui-manchette/dist/types';
+import type {
+  OperationalPoint,
+  SpaceTimeChartProps,
+} from '@osrd-project/ui-spacetimechart/dist/lib/types';
+
+import { BASE_OP_HEIGHT, MAX_TIME_WINDOW, MAX_ZOOM_X, MIN_ZOOM_X } from './consts';
+import { getHeightWithoutLastWaypoint, msToS } from './utils';
+
+export const calcOperationalPointsToDisplay = (
+  operationalPoints: OperationalPointType[],
+  height: number,
+  isProportional: boolean,
+  zoom: number
+): StyledOperationalPointType[] => {
+  if (!isProportional || operationalPoints.length === 0) {
+    // For non-proportional display, we always display all the operational points:
+    return operationalPoints.map((op) => ({ ...op, styles: {}, display: true }));
```

Comment:
styles can be undefined, it would be better to not defined it. Same for the other returns below
```suggestion
    return operationalPoints.map((op) => ({ ...op, display: true }));
```

Reply:
yes done it